### ISSUE TITLE
assert_redirected_to behaving oddly with Rails 6

### ISSUE THREAD

**Issue by brandonhilkert** [turbolinks/turbolinks-rails#44]
We have a bunch of tests that make `xhr` requests and used `assert_redirected_to`. 

Attempting to update to Rails 6, the same tests that passed before, fail with:

```
Schools::Parents::UploadsControllerTest#test_requires_superadmin [/Users/bhilkert/code/bark/test/controllers/schools/parents/uploads_controller_test.rb:47]:
Expected response to be a <3XX: redirect>, but was a <200: OK>
Response body: Turbolinks.clearCache()
Turbolinks.visit("http://bark.test/schools/dashboard", {"action":"replace"})

rails test test/controllers/schools/parents/uploads_controller_test.rb:39
```

It should use [Turbolink's version of `assert_redirected_to`](https://github.com/turbolinks/turbolinks-rails/blob/master/lib/turbolinks/assertions.rb#L13), however it's not because [the check](https://github.com/turbolinks/turbolinks-rails/blob/master/lib/turbolinks/assertions.rb#L6) is return false because `response.content_type` from Rails 6 is `text/javascript; charset=utf-8` instead of the checked condition (`text/javascript`).

Is this the expected response type for Rails 6 and if so, should this be adjusted to check for the presence of `text/javascript` instead of the exact string? If so, I can make a PR, but just want to confirm we don't have something configured wrong.

**Comment by georgeclaghorn** [comment]
This was fixed in #43. We need to cut a new release of this gem, but until we do, you can bundle from GitHub.

**Comment by brandonhilkert** [comment]
Thanks! Start going down the path of `ActionDispatch::Response.return_only_media_type_on_content_type = true` and then saw the new deprecation. 

```
DEPRECATION WARNING: Rails 6.1 will return Content-Type header without modification. If you want just the MIME type, please use `#media_type` instead. (called from assert_redirected_to at (pry):1)
